### PR TITLE
Fixed issue where File.createNewFile() sometimes caused an IOException

### PR DESCRIPTION
**Pull request by vrublack**
This occurred on my Nexus 5X sometimes although the operation worked on retries.

Fixed by retrying the operation a set number of times before it fails.

(see issue #4782)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/libanki/Utils.java [hunk 1]
```diff
@@ -692,38 +692,55 @@ public static void unzipFiles(ZipFile zipFile, String targetDirectory, String[]
...
-            long sizeKb = sizeBytes / 1024;
-            long speedKbSec = 0;
-            if (endTimeMillis != startTimeMillis) {
-                speedKbSec = sizeKb * 1000 / (endTimeMillis - startTimeMillis);
+        Timber.d("Creating new file... = %s", destination);
+        // sometimes this fails and works on retries (hardware issue?)
+        final int retries = 5;
+        int retryCnt = 0;
+        boolean created = false;
+        while (!created && retryCnt++ < retries) {
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/libanki/Utils.java]: Is it possible to leave the existing function how it is and handle the retry logic inside a separate function? I think that will be easier to maintain.

E.g. you could split into two functions

```c++
public static void writeToFile(InputStream source, String destination) throws IOException {
    // call writeToFileImpl() in a while loop
}
```

```c++
public static void writeToFileImpl(InputStream source, String destination) throws IOException {
    // the current code in master
}
```

**Comment by timrae** [comment]
Sorry for the delay on reviewing this btw, I don't have much time for AnkiDroid recently

File: AnkiDroid/src/main/java/com/ichi2/libanki/Utils.java (hunk 1 above)
Comment (vrublack) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Utils.java]: Okay did that

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/libanki/Utils.java]: Looks clean enough to me. I have one big change in my AnkiDroid work queue I'd like to dispatch before looking at this, but afterwards I'll pull this, independently test, and post back

**Comment by timrae** [comment]
Thank you @vrublack

**Pull request merged**